api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java

```diff
@@ -442,20 +443,25 @@ public void testSubmitTermsOfService() {
   }
 
   @Test
-  public void testSyncDuccVersionStatus_correctVersion() {
-    final DbUser user = userDao.findUserByUsername(USERNAME);
-    user.setDuccAgreement(signDucc(user, accessModuleService.getCurrentDuccVersion()));
-    userDao.save(user);
-
-    userService.syncDuccVersionStatus(user, Agent.asSystem());
-
-    verify(accessModuleService, never()).updateCompletionTime(any(), any(), any());
+  public void testSyncDuccVersionStatus_correctVersions() {
+    providedWorkbenchConfig.access.currentDuccVersions = ImmutableList.of(3, 4, 5);
+
+    providedWorkbenchConfig.access.currentDuccVersions.forEach(
+        version -> {
+          final DbUser user = userDao.findUserByUsername(USERNAME);
+          user.setDuccAgreement(signDucc(user, version));
+          userDao.save(user);
```

Comment:
```suggestion
          DbUser user = userDao.findUserByUsername(USERNAME);
          user.setDuccAgreement(signDucc(user, version));
          user = userDao.save(user);
```
Use the returned instance for further operations as the save operation might have changed the entity instance completely